app/models/generic_phrase.rb

```diff
@@ -46,16 +46,27 @@ def self.most_frequent_co_occurring(generic_phrase, start_date, end_date)
                                           .pluck("mentions.survey_answer_id")
 
     generic_phrases_for_survey_answers = GenericPhrase
-                                           .joins(:verb, :adjective, phrase_generic_phrases: [{ phrase: :mentions }])
+                                           .joins(phrase_generic_phrases: [{ phrase: :mentions }])
                                            .where("mentions.survey_answer_id" => survey_answers_for_generic_phrase)
-                                           .group("mentions.survey_answer_id, verbs.name, adjectives.name")
-                                           .order("concat(verbs.name, '-', adjectives.name) asc")
-                                           .pluck("mentions.survey_answer_id", "concat(verbs.name, '-', adjectives.name)")
-
-    generic_phrase_combinations = co_occurring_generic_phrases(generic_phrases_for_survey_answers)
-    generic_phrase_combinations
-        .each_with_object({}) { |(generic_phrase1, generic_phrase2), hsh| hsh["#{generic_phrase1}-#{generic_phrase2}"] = generic_phrase_combinations.count([generic_phrase1, generic_phrase2]) }
-        .sort_by { |generic_phrase_pair, occurrences| [-occurrences, generic_phrase_pair] }
+                                           .group("mentions.survey_answer_id, phrase_generic_phrases.generic_phrase_id")
+                                           .pluck("mentions.survey_answer_id", "phrase_generic_phrases.generic_phrase_id")
+
+    unique_generic_phrase_ids = generic_phrases_for_survey_answers.map { |_, id, _| id }.uniq
+    generic_phrases_by_id = generic_phrase_lookup_for_ids(unique_generic_phrase_ids)
+
+    generic_phrase_combinations = co_occurring_generic_phrases(generic_phrases_for_survey_answers, generic_phrase).map do |id1, id2|
+      other_phrase_id = id1 == generic_phrase.id ? id2 : id1
+      [generic_phrase.id, other_phrase_id]
+    end
+
+    generic_phrase_pair_counts = generic_phrase_combinations
+      .each_with_object({}) do |(id1, id2), hsh|
+        hsh[[id1, id2]] = generic_phrase_combinations.count([id1, id2])
```

Comment:
I'm guessing line 59/59 ensures that the order of `[id1, id2]` is always consistent? Ie so that we don't get `[1,2]` counted differently to `[2,1]` (or that if it does, that's the desired behaviour?)

Reply:
Actually scrub that, figured it out

Reply:
Cool. Just for the benefit of our future selves, the reason for line 59 is to ensure that the pairs are always displayed later on with the generic phrase we've selected first, and the other generic phrase second, so that we only ever link to a generic phrase which is different from the current generic phrase page we're on.